### ISSUE TITLE
Undefined reference to OSMPCNetworkProxy_* when building for dSpace Scalexio

### ISSUE THREAD

**Issue by stanislavbraun** [OpenSimulationInterface/osi-sensor-model-packaging#26]
Build of  OSMPCNetworkProxy fails for dSpace Scalexio with the following error:

```
Creating application image file ...
dsfmu_mdlfunctions.o86: In function `OSMPCNetworkProxy_DSFMUInitMdl':
dsfmu_mdlfunctions.cpp:(.text+0x5b): undefined reference to `OSMPCNetworkProxy_fmi2Instantiate'
dsfmu_mdlfunctions.o86: In function `OSMPCNetworkProxy_DSFMUStartMdl':
dsfmu_mdlfunctions.cpp:(.text+0x111): undefined reference to `OSMPCNetworkProxy_fmi2SetupExperiment'
dsfmu_mdlfunctions.cpp:(.text+0x143): undefined reference to `OSMPCNetworkProxy_fmi2SetBoolean'
dsfmu_mdlfunctions.cpp:(.text+0x159): undefined reference to `OSMPCNetworkProxy_fmi2EnterInitializationMode'
dsfmu_mdlfunctions.cpp:(.text+0x173): undefined reference to `OSMPCNetworkProxy_fmi2ExitInitializationMode'
dsfmu_mdlfunctions.cpp:(.text+0x2f9): undefined reference to `OSMPCNetworkProxy_fmi2Reset'
dsfmu_mdlfunctions.o86: In function `OSMPCNetworkProxy_DSFMUStep':
dsfmu_mdlfunctions.cpp:(.text+0x37e): undefined reference to `OSMPCNetworkProxy_fmi2SetInteger'
dsfmu_mdlfunctions.cpp:(.text+0x3c8): undefined reference to `OSMPCNetworkProxy_fmi2DoStep'
dsfmu_mdlfunctions.cpp:(.text+0x3fa): undefined reference to `OSMPCNetworkProxy_fmi2GetInteger'
dsfmu_mdlfunctions.cpp:(.text+0x42c): undefined reference to `OSMPCNetworkProxy_fmi2GetBoolean'
dsfmu_mdlfunctions.o86: In function `OSMPCNetworkProxy_DSFMUProvideInputs':
dsfmu_mdlfunctions.cpp:(.text+0x754): undefined reference to `OSMPCNetworkProxy_fmi2SetInteger'
dsfmu_mdlfunctions.o86: In function `OSMPCNetworkProxy_DSFMUGetOutputs':
dsfmu_mdlfunctions.cpp:(.text+0x844): undefined reference to `OSMPCNetworkProxy_fmi2GetInteger'
dsfmu_mdlfunctions.cpp:(.text+0x876): undefined reference to `OSMPCNetworkProxy_fmi2GetBoolean'
dsfmu_mdlfunctions.o86: In function `OSMPCNetworkProxy_DSFMUProvideParametersInit':
dsfmu_mdlfunctions.cpp:(.text+0xa24): undefined reference to `OSMPCNetworkProxy_fmi2SetBoolean'
dsfmu_mdlfunctions.o86: In function `OSMPCNetworkProxy_DSFMUStopMdl':
dsfmu_mdlfunctions.cpp:(.text+0xbec): undefined reference to `OSMPCNetworkProxy_fmi2Terminate'
collect2.exe: error: ld returned 1 exit status
OPUS MAKE: Shell line exit status 1. Stop.
Linking application image file "OSMPCNetworkProxy.x86" failed
Make failed
The make phase failed for the following application process: OSMPCNetworkProxy.
==== Build process aborted ===

```
This issue can be solved by adding

`#define FMI2_FUNCTION_PREFIX OSMPCNetworkProxy_`

in the .h file of the source code for OSMPCNetworkProxy

**Comment by pmai** [comment]
Sorry, this got missed in the transition from our internal build systems to the CMake public build: The prefix is only allowed to be defined for the sources in the source FMU, not for the sources that the binary FMUs are built from; this was re-added at a later stage by our build system, but doing this portably in Cmake is not simple to do, so I have switched to always include the prefix definition in the header files, but disabling it for cmake-based binary builds (changes are contained in PR #27).

**Comment by pmai** [comment]
Closing as fixed by #27.
